### PR TITLE
Conversion tracking datastore

### PR DESCRIPTION
**Pull request by zutigrm**
## Summary

Addresses issue:

- #8615

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/datastore/settings.js [hunk 1]
```diff
@@ -139,60 +140,112 @@ export async function submitChanges( { select, dispatch } ) {
...
+	if ( error ) {
+		return error;
+	}
+
+	await API.invalidateCache( 'modules', 'analytics-4' );
+
+	return {};
+}
+
+const saveSettings = async ( select, dispatch ) => {
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics-4/datastore/settings.js:168]: Are there any reasons to use an anonymous function style for this top-level function? If not, let's then use the standard declaration. The same applies for other top-level functions written in the arrow function style.

File: assets/js/modules/ads/datastore/settings.js [hunk 2]
```diff
@@ -38,12 +39,38 @@ export const INVARIANT_INVALID_CONVERSION_ID =
...
+	const haveConversionTrackingSettingsChanged =
+		select( CORE_SITE ).haveConversionTrackingSettingsChanged();
+
 	// This action shouldn't be called if settings haven't changed,
 	// but this prevents errors in tests.
-	if ( select( MODULES_ADS ).haveSettingsChanged() ) {
-		const { error } = await dispatch( MODULES_ADS ).saveSettings();
-		if ( error ) {
-			return { error };
+	if ( haveSettingsChanged || haveConversionTrackingSettingsChanged ) {
```
Comment (eugene-manuilov) [inline assets/js/modules/ads/datastore/settings.js:48]: Same as in the analytics settings. No need to have multi-level nesting here. Just return early when necessary.

File: assets/js/googlesitekit/datastore/site/conversion-tracking.js [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+	/**
+	 * Indicates whether saving the settings is currently in progress.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {boolean} True if the settings are being saved, false otherwise.
+	 */
+	isDoingSaveConversionTrackingSettings( state ) {
```
Comment (eugene-manuilov) [inline assets/js/googlesitekit/datastore/site/conversion-tracking.js:159]: This one isn't needed as far as I see. Let's remove this selector.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @zutigrm. A few comments from me.

File: assets/js/modules/analytics-4/datastore/settings.js [hunk 4]
```diff
@@ -139,60 +140,112 @@ export async function submitChanges( { select, dispatch } ) {
...
+	const haveSettingsChanged =
+		select( MODULES_ANALYTICS_4 ).haveSettingsChanged();
+	const haveConversionTrackingSettingsChanged =
+		select( CORE_SITE ).haveConversionTrackingSettingsChanged();
+
+	if ( haveSettingsChanged || haveConversionTrackingSettingsChanged ) {
```
[the remark this answers is not in the document]
Comment (zutigrm) [reply assets/js/modules/analytics-4/datastore/settings.js:174]: Oh indeed that is a great point, I got carried away with when new settings should be saved 🤦

File: assets/js/googlesitekit/datastore/site/conversion-tracking.js (hunk 3 above)
Comment (zutigrm) [reply assets/js/googlesitekit/datastore/site/conversion-tracking.js:159]: We are going to most likely add issue for including the Conversion Tracking into the setup Ads view, but yeah I guess we can just include this selector together with other validate submit/can submit updates in that issue once added. Removed for now

**Comment by zutigrm** [comment]
@eugene-manuilov THanks for the feedback, PR updated

**Review by eugene-manuilov — APPROVED** [review]
Thanks, @zutigrm. LGTM

**Pull request merged**